server/plpgsql/interpreter_stack.go

```diff
@@ -26,8 +27,15 @@ import (
 
 // InterpreterVariable is a variable that lives on the stack.
 type InterpreterVariable struct {
+	Record sql.Schema
+	Type   *pgtypes.DoltgresType
+	Value  any
+}
+
+// InterpreterVariableReference is a reference to a variable that lives on the stack.
```

Comment:
Can you explain some more in these docs about the intent of `InterpreterVariableReference` and `InterpreterVariable` and how they are supposed to be used? Seems likely future code readers will need some more context here. 

From looking at current usage, it seems like `InterpreterVariable` could potentially be unexported to `interpreterVariable` and is not intended to ever be returned directly, but more of an internal record keeping for the variables. Documenting that, if that's correct, would be helpful for future people to more quickly understand the intent of these types and keep them used in the way you want them used.